sae_lens/training/activations_store.py

```diff
@@ -395,10 +395,22 @@ def load_cached_activation_dataset(self) -> Dataset | None:
 
         return activations_dataset
 
+    def set_norm_scaling_factor_if_needed(self):
+        if self.normalize_activations == "expected_average_only_in":
+            self.estimated_norm_scaling_factor = self.estimate_norm_scaling_factor()
+
     def apply_norm_scaling_factor(self, activations: torch.Tensor) -> torch.Tensor:
+        if self.estimated_norm_scaling_factor is None:
+            raise ValueError(
+                "estimated_norm_scaling_factor is not set, call set_norm_scaling_factor_if_needed() first"
+            )
         return activations * self.estimated_norm_scaling_factor
 
     def unscale(self, activations: torch.Tensor) -> torch.Tensor:
+        if self.estimated_norm_scaling_factor is None:
+            raise ValueError(
+                "estimated_norm_scaling_factor is not set, call set_norm_scaling_factor_if_needed() first"
+            )
```

Comment:
before, if calling either of these before setting `self.estimated_norm_scaling_factor`, it would work, which is _kind of_ a (user-caused) bug. this way that can't happen